### ISSUE TITLE
Add a canary at the end of thread struct

### ISSUE THREAD

**Issue by bmarcot** [bmarcot/vega#11]
```
-------------
struct thread {
  struct mthread mthread;
  unsinged int id
  char canary[4] = "cana";
};
-------------
^
| interrupt stack
| growing downward
```

On an exception, it should be possible in some cases to catch a stack overflow in the interrupt stack by checking if a canary has been clobbered.

**Comment by bmarcot** [comment]
Also, same idea can be used for stack profiling when building with `--interrupt-stack-profiling` and `--thread-stack-profiling`. Check how much pattern have been clobbered.

**Comment by bmarcot** [comment]
Added in f0a3679

**Comment by bmarcot** [comment]
Fixed in f0a36791bc462f8129ead6b0bb11c35adf95e1cf
